contrib/internet-urllib3.py

```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env python
+
+import urllib3
+import certifi
+
+def isInternetConnected(url="www.ietf.org"):
+  result = False
+  http = urllib3.PoolManager(
+    cert_reqs='CERT_REQUIRED', # Force certificate check.
+    ca_certs=certifi.where(),  # Path to the Certifi bundle.
+  )
+  try:
+    r = http.request('HEAD', 'https://' + url)
+    result = True
+  except Exception as e:  # urllib3.exceptions.SSLError
+    result = False
```

Comment:
OK.  In extremely secure environments (banks, government, etc), sometimes firewalls and proxies block web access.  But that is a strange case - why try to sync email with offlineimap from inside a bank local network.  If web access is blocked, fall back to authenticate the cert of the imap server - yet, imap servers are not required to even have a vaild cert, so the best is to connect over https to a known internet web server and use the authenticity of the web server cert as proof that the internet is connected.